backend_jobs/ingestion_pipeline/pipeline_lib/image_provider_interface.py

```diff
@@ -0,0 +1,99 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ """
+ # pylint: disable=pointless-string-statement
+
+from abc import ABC, abstractmethod
+import apache_beam as beam
+
+"""
+Each image provider that is added to the platform will inherit from the ImageProviders class.
+This class is incharge of:
+* returning images from the provider
+* calculting the num of baches we want to run in parallel
+* extarcting additional metadata.
+"""
+
+class ImageProvider(ABC, beam.DoFn):
+    """
+    ImageProvider is an interface that all the image ingestion provider's inherit from
+    """
+
+    @abstractmethod
+    def get_images(self, num_of_batches, num_of_images, query_arguments):
+        """
+        This function is incharge of calling an API/Image provider source
+        and receives a list of images
+        Args:
+            num_of_batches: the number of the batches we want to run in parallel
```

Comment:
i think you want page_number here instead

(this class has no concept of parallelism or apache beam)

Reply:
fixed to:
 num_of_page: The number of page we want to retrieve images from.